recipes/armadillo/all/conanfile.py

```diff
@@ -195,89 +190,75 @@ def requirements(self):
+        tc.cache_variables["USE_SYSTEM_HDF5"] = False
+        tc.cache_variables["USE_SYSTEM_ARPACK"] = self.options.use_arpack
+        tc.cache_variables["USE_SYSTEM_SUPERLU"] = self.options.use_superlu
+        tc.cache_variables["USE_SYSTEM_OPENBLAS"] = False
+        tc.cache_variables["USE_SYSTEM_FLEXIBLAS"] = self.options.use_blas == "system_flexiblas"
+        tc.cache_variables["ALLOW_FLEXIBLAS_LINUX"] = self.options.use_blas == "system_flexiblas" and self.settings.os == "Linux"
+        tc.cache_variables["ALLOW_OPENBLAS_MACOS"] = self.options.use_blas == "openblas" and self.settings.os == "Macos"
+        tc.cache_variables["OPENBLAS_PROVIDES_LAPACK"] = self.options.use_lapack == "openblas"
+        tc.cache_variables["ALLOW_BLAS_LAPACK_MACOS"] = self.options.use_blas != "framework_accelerate"
+        tc.cache_variables["BUILD_SHARED_LIBS"] = self.options.shared
+        tc.cache_variables["BUILD_SMOKE_TEST"] = False
+        tc.generate()
+
+        deps = CMakeDeps(self)
+        deps.generate()
 
     def source(self):
-        tools.get(
+        get(self,
             **self.conan_data["sources"][self.version],
             strip_root=True,
-            destination=self._source_subfolder,
-            filename="{name}-{version}.tar.xz".format(
-                name=self.name, version=self.version
-            ),
+            filename="f{self.name}-{self.version}.tar.xz"
         )
 
-    def _patch_sources(self):
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            tools.patch(**patch)
-
     def build(self):
-        self._patch_sources()
-        cmake = self._configure_cmake()
+        apply_conandata_patches(self)
+        replace_in_file(
+            self,
+            os.path.join(self.source_folder, "CMakeLists.txt"),
+            "include(CheckLibraryExists)",
+            "include(CheckLibraryExists)\nset(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS on)",
+            strict=False,
+        )
+
+        cmake = CMake(self)
+        cmake.configure()
         cmake.build()
 
     def package(self):
-        self.copy("LICENSE.txt", dst="licenses", src=self._source_subfolder)
-        self.copy("NOTICE.txt", dst="licenses", src=self._source_subfolder)
-        cmake = self._configure_cmake()
+        cmake = CMake(self)
         cmake.install()
-        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
-        tools.rmdir(os.path.join(self.package_folder, "share"))
+
+        copy(self, "LICENSE.txt", dst=os.path.join(self.package_folder, "licenses"), src=self.source_folder)
+        copy(self, "NOTICE.txt", dst=os.path.join(self.package_folder, "licenses"), src=self.source_folder)
+        rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
+        rmdir(self, os.path.join(self.package_folder, "share"))
 
     def package_info(self):
         self.cpp_info.libs = ["armadillo"]
-        self.cpp_info.names["pkg_config"] = "armadillo"
```

Comment:
This should be a set property since I can see you removed the installed files

Reply:
Is this necessary? I've removed this line because it's just defining the default - my understanding is that the `pkg_config_name` property should use the package name by default too?

Reply:
It's necessary because it's a way to say "I saw the project and they set that name explicitly". So in that case, please, add `self.cpp_info..set_property("pkg_config_name", "armadillo")`

Reply:
This is a CCI convention to make lives easier when reviewing... if it's there then its official and we dont need to double check.

In general, you should not need this